Add basics/account-data/steel

0xCipherCoder opened the pull request:
- Updated anchor code as per guidelines and updated version
- Added an account data example using the [Steel framework](https://github.com/regolith-labs/steel) using [Bankrun](https://kevinheavey.github.io/solana-bankrun/) tests.

heyAyushh commented:
@0xCipherCoder  #198, Merge main to your branch again.

heyAyushh commented on Cargo.toml:
https://github.com/solana-developers/program-examples/blob/main/CONTRIBUTING.md#general-coding-and-writing-guidelines

0xCipherCoder replied:
What is required to remove? @heyAyushh

0xCipherCoder replied:
Got it.

0xCipherCoder commented:
> @0xCipherCoder #198, Merge main to your branch again.

Thanks for the review! @heyAyushh I have merged the main branch and removed steel from `cargo.toml` file

0xCipherCoder commented:
@heyAyushh I have fixed clippy and rustfmt related formatting and style issues. Please review it once.

0xCipherCoder commented:
> @heyAyushh I have fixed clippy and rustfmt related formatting and style issues. Please review it once.

@heyAyushh I have fixed clippy issues as well.
